view/desktop/src/layouts/AppLayout.tsx

```diff
@@ -10,55 +10,306 @@ import { Resizable, ResizablePanel } from "../components/Resizable";
+  const primarySideBarSetWidth = useLayoutStore((state) => state.primarySideBar.setWidth);
+  const primarySideBarGetWidth = useLayoutStore((state) => state.primarySideBar.getWidth);
+
+  const secondarySideBarVisibility = useLayoutStore((state) => state.secondarySideBar.visibility);
+  const secondarySideBarSetWidth = useLayoutStore((state) => state.secondarySideBar.setWidth);
+  const secondarySideBarGetWidth = useLayoutStore((state) => state.secondarySideBar.getWidth);
+
+  const bottomPaneVisibility = useLayoutStore((state) => state.bottomPane.visibility);
+  const bottomPaneSetHeight = useLayoutStore((state) => state.bottomPane.setHeight);
+  const bottomPaneGetHeight = useLayoutStore((state) => state.bottomPane.getHeight);
+
+  return (
+    <Resizable
+      onDragEnd={(sizes) => {
+        const [primarySideBarWidth, _, secondarySideBarWidth] = sizes;
+        primarySideBarSetWidth(primarySideBarWidth);
+        secondarySideBarSetWidth(secondarySideBarWidth);
+      }}
+    >
+      <ResizablePanel
+        minSize={100}
+        preferredSize={primarySideBarGetWidth()}
+        snap
+        visible={primarySideBarVisibility}
+        className="select-none"
+      >
+        <PrimarySideBar />
+      </ResizablePanel>
+      <ResizablePanel>
+        <Resizable
+          vertical
+          onDragEnd={(sizes) => {
+            const [_, bottomPaneHeight] = sizes;
+            bottomPaneSetHeight(bottomPaneHeight);
+          }}
+        >
+          <ResizablePanel>
+            <MainContent />
+          </ResizablePanel>
+          <ResizablePanel preferredSize={bottomPaneGetHeight()} snap minSize={100} visible={bottomPaneVisibility}>
+            <BottomPaneContent />
+          </ResizablePanel>
+        </Resizable>
+      </ResizablePanel>
+      <ResizablePanel
+        minSize={100}
+        preferredSize={secondarySideBarGetWidth()}
+        snap
+        visible={secondarySideBarVisibility}
+        className="select-none"
+      >
+        <SecondarySideBar />
+      </ResizablePanel>
+    </Resizable>
+  );
+};
+
+const JustifyLayout = () => {
+  const primarySideBarVisibility = useLayoutStore((state) => state.primarySideBar.visibility);
+  const primarySideBarSetWidth = useLayoutStore((state) => state.primarySideBar.setWidth);
+  const primarySideBarGetWidth = useLayoutStore((state) => state.primarySideBar.getWidth);
+
+  const secondarySideBarVisibility = useLayoutStore((state) => state.secondarySideBar.visibility);
+  const secondarySideBarSetWidth = useLayoutStore((state) => state.secondarySideBar.setWidth);
+  const secondarySideBarGetWidth = useLayoutStore((state) => state.secondarySideBar.getWidth);
+
+  const bottomPaneVisibility = useLayoutStore((state) => state.bottomPane.visibility);
+  const bottomPaneSetHeight = useLayoutStore((state) => state.bottomPane.setHeight);
+  const bottomPaneGetHeight = useLayoutStore((state) => state.bottomPane.getHeight);
+
+  return (
```

Comment:
It looks like this section of code is repeated more than twice. Can we consider some form of abstraction to adhere to the DRY principle?

Reply:
Changed it in the new commit

Reply:
LGTM